README.md

```diff
@@ -1,3 +1,280 @@
+require('bootstrap-loader');
+```
+
+Or add `bootstrap-loader` as an entry point in your webpack config:
+
+```js
+entry: [ 'bootstrap-loader', './app' ]
+```
+
+Config is optional. It can be placed in root dir with name `.bootstraprc`. You can write it in `YAML` or `JSON` formats.
+
+```yaml
+---
+# You can use comments here
+useFlexbox: true
+
+styleLoaders:
+  - style
+  - css
+  - sass
+
+styles:
+  normalize: true
+  print: true
+
+scripts:
+  alert: true
+  button: true
+```
+
+```json
+{
+  // And JSON comments also!
+  "useFlexbox": true,
+
+  "styleLoaders": ["style", "css", "sass"],
+
+  "styles": {
+    "normalize": true,
+    "print": true
+  },
+
+  "scripts": {
+    "alert": true,
+    "button": true
+  }
+}
+```
+
+If no config provided, default one for Bootstrap 3 will be used.
+
+## Examples
+
+Check out example apps:
+
+* Basic usage: https://github.com/shakacode/bootstrap-loader-example
+* With CSS Modules: https://github.com/shakacode/bootstrap-loader-css-modules-example
+
+## Options
+
+Here are options for Bootstrap 3 & 4.
+
+### Bootstrap 3
+
+#### `loglevel`
+
+Default: `disabled`
+
+Outputs debugging info. In case you have troubles with loader you can set this option to `debug` and send us output form console.
+
+```yaml
+loglevel: debug
+```
+
+#### `bootstrapVersion`
+
+Default: `3`
+
+Major version of Bootstrap. Can be 3 or 4.
+
+```yaml
+bootstrapVersion: 3
+```
+
+#### `styleLoaders`
+
+Default: `[ 'style', 'css', 'sass' ]`
+
+Array of webpack loaders. `sass-loader` is required, order matters.
+
+```yaml
+styleLoaders:
+  - style
+  - css
+  - sass
+
+# You can apply loader params here:
+  - sass?outputStyle=expanded
+```
+
+#### `extractStyles`
+
+Default: `false`
+
+Extract styles to stand-alone css file using `extract-text-webpack-plugin`.
+
+```yaml
+extractStyles: false
+
+# Different settings for different environments can be used,
+# It depends on value of NODE_ENV environment variable
+env:
+  development:
+    extractStyles: false
+  production:
+    extractStyles: true
+```
+
+This param can also be set to `true` in webpack config:
+
+```js
+entry: [ 'bootstrap-loader/extractStyles', './app' ]
+```
+
+#### `preBootstrapCustomizations`
+
+Default: `disabled`
+
+Customize Bootstrap variables that get imported before the original Bootstrap variables. Thus, derived Bootstrap variables can depend on values from here. See the Bootstrap [`_variables.scss`](https://github.com/twbs/bootstrap-sass/blob/master/assets/stylesheets/bootstrap/_variables.scss) file for examples of derived Bootstrap variables.
+
+```yaml
+preBootstrapCustomizations: ./path/to/bootstrap/pre-customizations.scss
+```
+
+#### `bootstrapCustomizations`
+
+Default: `disabled`
+
+This gets loaded after bootstrap variables is loaded. Thus, you may customize Bootstrap variables based on the values established in the Bootstrap [`_variables.scss`](https://github.com/twbs/bootstrap-sass/blob/master/assets/stylesheets/bootstrap/_variables.scss) file.
```

Comment:
Note, if `bootstrap` did not have derived values, it would not be necessary to have two config files for customizing bootstrap values. 

If you want your bootstrap override value to apply to derived variable values, then place your customizations in `preBootstrapCustomizations`. If you want to be sure your changes don't affect other derived values, place the changes in `bootstrapCustomizations`.

If you are not sure, you can probably simply use `preBootstrapCustomizations` and, if you have issues, see  [`_variables.scss`](https://github.com/twbs/bootstrap-sass/blob/master/assets/stylesheets/bootstrap/_variables.scss) for derived values.